### PR TITLE
Simplify modal component

### PR DESCRIPTION
**Pull request by nmattia**
This simplifies the modal component code by handling more of the logic through lit constructs.

### CODE REVIEW

File: src/frontend/src/components/modal.ts [hunk 1]
```diff
@@ -7,78 +7,47 @@ import { withRef } from "../utils/lit-html";
...
-        <button
-          class="c-modal__close"
-          aria-label="Close Modal"
-          ${ref(closeButtonRef)}
-        >
-          &times;
-        </button>
-        ${slot}
-        <div class="c-modal__footer">
+    <dialog class="c-modal" aria-modal ${ref(modalElement)}>
```
Comment (nmattia) [inline src/frontend/src/components/modal.ts:24]: @peterpeterparker `aria-modal` is equivalent to `aria-modal="true"`, right?

Comment (peterpeterparker) [reply src/frontend/src/components/modal.ts:24]: Good question, not sure. The default value is `false`.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal

**Review by peterpeterparker — APPROVED** [review]
Rubber stamp

File: src/frontend/src/components/modal.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/components/modal.ts:24]: > Boolean attributes are considered to be true if they're present on the element at all.
> https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute

all good

Comment (peterpeterparker) [reply src/frontend/src/components/modal.ts:24]: Great!

File: src/frontend/src/components/modal.ts [hunk 2]
```diff
@@ -7,78 +7,47 @@ import { withRef } from "../utils/lit-html";
...
-  submitButtonRef: Ref<HTMLButtonElement>;
-}): HTMLDialogElement => {
+  submit: () => void;
+}) => {
+  const modalElement: Ref<HTMLDialogElement> = createRef();
+
+  const closeModal = () =>
+    withRef(modalElement, (modalElement) => {
+      modalElement.close();
+      modalElement.remove();
```
Comment (nmattia) [inline src/frontend/src/components/modal.ts:20]: this means that clicking the "X" button removes the element, but submitting doesn't. Will need to figure out what we want exactly.

**Pull request merged**